### PR TITLE
Switch to bitcoind generic implementation

### PR DESCRIPTION
**Pull request by lutangar**
Hey guys, looking for a quick review, even if you don't know wtf is this, you may check for typo and such.

### CODE REVIEW

File: metrics.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    name: 'chain_difficulty',
+    help: 'The proof-of-work difficulty as a multiple of the minimum difficulty',
+});
+const walletVersionMetric = new Gauge({
+    name: 'wallet_version',
+    help: 'the wallet version',
+    labelNames: ['symbol']
+});
+const walletBalanceMetric = new Gauge({
+    help: `the total balance of the wallet`,
```
Comment (pocky) [inline metrics.js]: inverse "help" <-> "name" *(^_^)*

File: README.md [hunk 2]
```diff
@@ -1,17 +1,101 @@
...
+> `bitcoind-prometheus-exporter` uses the `bitcoind` **JSON-RPC** API under the hood and need those credentials:
+> ```
+>rpcuser=test
+>rpcpassword=1cf98b57-5i09-4fa1-9c07-2e28cb2cb47b
+>```
+
+## Usage with other wallets
+The following environment variables are available, just change them to suit your needs:
+> ```
+>symbol=DASH
```
Comment (pocky) [inline README.md]: `symbol=BTC` ?

Comment (lutangar) [reply README.md]: This is under the *Usage with other wallets* section, that's why it is not BTC here, but maybe this is confusing?

File: README.md [hunk 3]
```diff
@@ -1,17 +1,101 @@
...
+Most important metrics are:
+* wallet total balance
+* wallet version
+* if the wallet is unlocked
+* available (spendable) balance for each managed addresses (and watched addresses)
+* best block time and index
+* ...
 
 ## Usage
+Edit the `.env` environment file to suit your need and run:
```
Comment (Goutte) [inline README.md]: need**s**

**Review by Goutte — APPROVED** [review]
Looks OK to me, but I'm no expert. :100:

File: README.md [hunk 4]
```diff
@@ -1,17 +1,101 @@
-# dashd-prometheus-exporter
-![dashd prometheus exporter logo](dashd-prometheus-exporter.png)
+# bitcoind-prometheus-exporter
+Bitcoind wallet metrics **Prometheus** exporter.
 
-Dash headless wallet daemon exporter for Prometheus.
+> `bitcoind-prometheus-exporter` is compatible with most **bitcoin** forks.
+
+Produce **wallet** metrics, own **addresses** but also **blockchain** infos.
```
Comment (josqu4red) [inline README.md]: "own" ?

File: README.md [hunk 5]
```diff
@@ -1,17 +1,101 @@
...
 ````
 
 >An easy hack could be to directly use your wallet conf to feed your exporter `env`:
 >```
->docker run --name my-exporter -p 3000:3000 -v /path/to/my/conf:/app/.env --link my-wallet lepetitbloc/dashd-prometheus-exporter
+>docker run --name my-exporter -p 3000:3000 -v /path/to/my/conf:/app/.env --link my-wallet lepetitbloc/bitcoind-prometheus-exporter
 >```
 
+## Example
+When visiting the metrics URL http://localhost:3000/metrics the following **metrics** are produced:
```
Comment (josqu4red) [inline README.md:52]: :100:

**Review by josqu4red — COMMENTED** [review]
Can't really comment on code, but :+1: from a higher level POV

File: README.md [hunk 6]
```diff
@@ -1,17 +1,101 @@
...
+# TYPE wallet_version gauge
+wallet_version{symbol="BTC"} 71000
+
+# HELP wallet_balance_total the total balance of the wallet
+# TYPE wallet_balance_total gauge
+wallet_balance_total{unconfirmed="2.7345",immature="0",confirmed="42.73453501"} 42.73453501
+
+# HELP wallet_transactions_total the total number of transactions in the wallet
+# TYPE wallet_transactions_total gauge
+wallet_transactions_total 77
```
Comment (lutangar) [inline README.md]: @josqu4red I'm wondering how these wallet metrics will play together when monitoring many wallets... Should the metric include some kind of identifier? Or would that be handle on the collect side?
For addresses balances, it seems pretty straightforward with the address as the label (even so there could be collision with the same address appearing on 2 different forks, but that would be very unlikely though).

**Comment by pocky** [comment]
👍 but it's important to improve documentation and README file :)

File: README.md (hunk 6 above)
Comment (lutangar) [reply README.md]: @josqu4red ping
>lutangar reviewed 11 days ago

**Pull request merged**